### PR TITLE
LPS-124170 Add AddSegmentsExperienceMVCActionCommandTest UpdateSegmentsExperienceMVCActionCommandTest.

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

Ass Integration tests for UpdateSegmentsExperienceMVCActionCommandTest and AddSegmentsExperienceMVCActionCommandTest

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:reevaluate:1157401_3968

**Comment by cgoncas** [comment]
ci:test.relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 1]
```diff
@@ -91,14 +91,22 @@ public void tearDown() {
 	}
 
 	@Test
-	public void testAddSegmentsExperiment() throws Exception {
+	public void testAddSegmentsExperience() throws Exception {
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:102]: 😲

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 2]
```diff
@@ -91,14 +91,22 @@ public void tearDown() {
...
 		String name = RandomTestUtil.randomString(10);
 
 		SegmentsEntry segmentsEntry = SegmentsTestUtil.addSegmentsEntry(
 			_group.getGroupId());
 
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:108]: Much clearer now 👏

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 3]
```diff
@@ -160,6 +162,78 @@ public void testAddSegmentsExperience() throws Exception {
...
 
+	@Test
+	public void testAddSegmentsExperienceWithoutLanguageIds() throws Exception {
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			_getMockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.addParameter(
+			"name", RandomTestUtil.randomString());
+
+		_addSegmentsExperience(mockLiferayPortletActionRequest);
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:174]: Is this called needed?

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 4]
```diff
@@ -160,6 +162,78 @@ public void testAddSegmentsExperience() throws Exception {
...
+		mockLiferayPortletActionRequest.addParameter(
+			"languageIds",
+			StringUtil.merge(
+				LocaleUtil.toLanguageIds(
+					new Locale[] {LocaleUtil.US, LocaleUtil.BRAZIL}),
+				StringPool.COMMA));
+		mockLiferayPortletActionRequest.addParameter(
+			"name", RandomTestUtil.randomString());
+
+		_addSegmentsExperience(mockLiferayPortletActionRequest);
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:225]: Is this called needed?

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 5]
```diff
@@ -160,6 +162,78 @@ public void testAddSegmentsExperience() throws Exception {
...
+				StringPool.COMMA));
+		mockLiferayPortletActionRequest.addParameter(
+			"segmentsEntryId",
+			String.valueOf(segmentsEntry.getSegmentsEntryId()));
+
+		_addSegmentsExperience(mockLiferayPortletActionRequest);
+	}
+
+	@Test
+	public void testAddSegmentsExperienceWithoutSegmentsEntryId()
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:210]: Super useful test!

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 6]
```diff
@@ -160,6 +162,78 @@ public void testAddSegmentsExperience() throws Exception {
...
+			_group.getGroupId());
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			_getMockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.addParameter(
+			"languageIds",
+			StringUtil.merge(
+				LocaleUtil.toLanguageIds(
+					new Locale[] {LocaleUtil.US, LocaleUtil.BRAZIL}),
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:200]: Should BRAZIL appear before US?

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 7]
```diff
@@ -160,6 +162,78 @@ public void testAddSegmentsExperience() throws Exception {
...
+		throws Exception {
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			_getMockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.addParameter(
+			"languageIds",
+			StringUtil.merge(
+				LocaleUtil.toLanguageIds(
+					new Locale[] {LocaleUtil.US, LocaleUtil.BRAZIL}),
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:220]: Should BRAZIL appear before US?

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java [hunk 8]
```diff
@@ -100,6 +105,12 @@ public void testAddSegmentsExperience() throws Exception {
 		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
 			_getMockLiferayPortletActionRequest();
 
+		mockLiferayPortletActionRequest.addParameter(
+			"languageIds",
+			StringUtil.merge(
+				LocaleUtil.toLanguageIds(
+					new Locale[] {LocaleUtil.US, LocaleUtil.BRAZIL}),
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:115]: Should BRAZIL appear before US?

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/UpdateSegmentsExperienceMVCActionCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+			new MockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, new ThemeDisplay());
+
+		mockLiferayPortletActionRequest.addParameter(
+			"languageIds",
+			StringUtil.merge(
+				LocaleUtil.toLanguageIds(
+					new Locale[] {LocaleUtil.US, LocaleUtil.BRAZIL}),
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/UpdateSegmentsExperienceMVCActionCommandTest.java:95]: Should BRAZIL appear before US?

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/UpdateSegmentsExperienceMVCActionCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+			"newName",
+			actualSegmentsExperience.getName(LocaleUtil.getSiteDefault()));
+
+		UnicodeProperties typeSettingsUnicodeProperties =
+			actualSegmentsExperience.getTypeSettingsUnicodeProperties();
+
+		Assert.assertEquals(
+			StringUtil.merge(
+				LocaleUtil.toLanguageIds(
+					new Locale[] {LocaleUtil.US, LocaleUtil.BRAZIL}),
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/UpdateSegmentsExperienceMVCActionCommandTest.java:129]: Should BRAZIL appear before US?

**Review by darquesdev — APPROVED** [review]
LGTM! Just minor comments

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java (hunk 3 above)
Comment (cgoncas) [reply modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java:174]: No it is not :)

File: modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java (hunk 4 above)
Comment (cgoncas) [reply modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/test/AddSegmentsExperienceMVCActionCommandTest.java]: No, it is not :)

**Comment by cgoncas** [comment]
Thanks @darquesdev for the review, I have. applied your feedback :)

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**